compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -2029,14 +2030,17 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
         self.expr_block(block)
     }
 
-    fn lower_array_length(&mut self, c: &AnonConst) -> hir::ArrayLen<'hir> {
+    fn lower_array_length_to_const_arg(&mut self, c: &AnonConst) -> &'hir hir::ConstArg<'hir> {
         match c.value.kind {
             ExprKind::Underscore => {
                 if self.tcx.features().generic_arg_infer() {
-                    hir::ArrayLen::Infer(hir::InferArg {
-                        hir_id: self.lower_node_id(c.id),
+                    let hir_id = self.lower_node_id(c.id);
+                    let ct_kind = hir::ConstArgKind::Infer(hir::InferArg {
+                        hir_id,
                         span: self.lower_span(c.value.span),
-                    })
+                    });
+                    // FIXME(void): should InferArg and ConstArg share the same hir_id?
+                    self.arena.alloc(hir::ConstArg { hir_id, kind: ct_kind })
```

Comment:
I don't think so, it would mean that there's both a `Node::Infer` and a `Node::ConstArg` for the same `hir_id`. The `NodeCollector` will visit the const arg and then visit the `InferArg` inside of it and wind up calling `insert` for both. I think we should probably not be storing an `InferArg` in `ConstArgKind::Infer` as `TyKind::Infer` also does not store an `InferArg`.

I imagine this is the cause of the ICEs in the CI failure

Reply:
> I think we should probably not be storing an `InferArg` in `ConstArgKind::Infer` as `TyKind::Infer` also does not store an `InferArg`.

Ah, I see, so we would have `ConstArgKind::Infer(Span)` and not need an additional `HirId`?

Reply:
yep :3